examples/cc3200/xively_firmware_updates/SFT/xi_sft_client.c

```diff
@@ -497,24 +501,26 @@ void xi_publish_file_info( xi_context_handle_t in_context_handle )
 {
     /* Let's try making a XI_FILE_INFO packet */
     cn_cbor_errback err;
-    cn_cbor* cb_map = cn_cbor_map_create( &err );
+    cn_cbor* cb_map = cn_cbor_map_create( CBOR_CONTEXT_PARAM_COMMA & err );
 
-    cn_cbor_map_put( cb_map, cn_cbor_string_create( "msgtype", &err ),
-                     cn_cbor_int_create( XI_FILE_INFO, &err ), &err );
+    cn_cbor_map_put( cb_map, cn_cbor_string_create( "msgtype" CBOR_CONTEXT_PARAM, &err ),
+                     cn_cbor_int_create( XI_FILE_INFO CBOR_CONTEXT_PARAM, &err ), &err );
 
-    cn_cbor_map_put( cb_map, cn_cbor_string_create( "msgver", &err ),
-                     cn_cbor_int_create( 1, &err ), &err );
+    cn_cbor_map_put( cb_map, cn_cbor_string_create( "msgver" CBOR_CONTEXT_PARAM, &err ),
+                     cn_cbor_int_create( 1 CBOR_CONTEXT_PARAM, &err ), &err );
 
-    cn_cbor* a        = cn_cbor_array_create( &err );
-    cn_cbor* cb_file1 = cn_cbor_map_create( &err );
-    cn_cbor_map_put( cb_file1, cn_cbor_string_create( "N", &err ),
-                     cn_cbor_string_create( firmware_filename, &err ), &err );
+    cn_cbor* a        = cn_cbor_array_create( CBOR_CONTEXT_PARAM_COMMA & err );
+    cn_cbor* cb_file1 = cn_cbor_map_create( CBOR_CONTEXT_PARAM_COMMA & err );
+    cn_cbor_map_put( cb_file1, cn_cbor_string_create( "N" CBOR_CONTEXT_PARAM, &err ),
+                     cn_cbor_string_create( firmware_filename CBOR_CONTEXT_PARAM, &err ),
+                     &err );
 
-    cn_cbor_map_put( cb_file1, cn_cbor_string_create( "R", &err ),
-                     cn_cbor_string_create( "-1", &err ), &err );
+    cn_cbor_map_put( cb_file1, cn_cbor_string_create( "R" CBOR_CONTEXT_PARAM, &err ),
```

Comment:
yes, and yes, it isn't an ordinary solution from the CN-CBOR C codec, but works and has to be used like this. The macro contains the coma. And the macro can be empty as well when no context is used. That is why it looks like this.